src/plugins/discover/public/application/main/services/discover_app_state_container.ts

```diff
@@ -143,6 +144,10 @@ export interface DiscoverAppState {
    * Breakdown field of chart
    */
   breakdownField?: string;
+  /**
+   * Customization for Lens vis in Histogram area
+   */
+  visContext?: ExternalVisContext;
```

Comment:
Naming:
- In Discover app state it's `visContext` as object
- In UnifiedHistogramContainer prop it's `externalVisContext` as object
- In saved search its `visContextJSON` as string

Reply:
Having a look at one of the problems. I evaluated moving the visContext to InternalStateContainer, preventing the evil URL syncing causing huge pain, and it seems the change detection works then. 

https://github.com/elastic/kibana/pull/175876/commits/3abe5e0e529e6ed5ab02d33093affe065c101114

https://github.com/elastic/kibana/assets/463851/f250cbdd-979c-4e75-9b91-87ea0983190a

note that this is just very quick and dirty, but looking at the state syncing I think it's showing a path forward